internal/datacoord/garbage_collector.go

```diff
@@ -178,103 +179,146 @@ func (gc *garbageCollector) work() {
 	}
 }
 
+// triggerRecycleFiles is used to trigger recycle files
+func (gc *garbageCollector) triggerRecycleFiles() {
+	select {
+	case gc.fileRecycleCh <- struct{}{}:
+	default:
+	}
+}
+
+// recycleFilesBackgroundTask is used to recycle unused files
+func (gc *garbageCollector) recycleFilesBackgroundTask() {
+	defer gc.wg.Done()
+	for {
+		select {
+		case <-gc.fileRecycleCh:
+			gc.recycleUnusedBinlogFiles()
+			gc.recycleUnusedIndexFiles()
+		case <-gc.closeCh:
+			log.Warn("garbage collector recycle files background task quit")
+			return
+		}
+	}
+}
+
 func (gc *garbageCollector) close() {
 	gc.stopOnce.Do(func() {
 		close(gc.closeCh)
 		gc.wg.Wait()
 	})
 }
 
-// scan load meta file info and compares OSS keys
+// recycleUnusedBinlogFiles load meta file info and compares OSS keys
 // if missing found, performs gc cleanup
-func (gc *garbageCollector) scan() {
+func (gc *garbageCollector) recycleUnusedBinlogFiles() {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	var (
-		total   = 0
-		valid   = 0
-		missing = 0
-	)
-	getMetaMap := func() (typeutil.UniqueSet, typeutil.Set[string]) {
```

Comment:
I think continuing with `getMetaMap` pattern may lead to better performance when compared to checking `IsLogExists`.

Reply:
getMetaMap can not be executed before `WalkWithPrefix`, otherwise if `WalkWithPrefix` executes too much time, snapshot generated by getMetaMap is too old, miss tolerance cannot protects valid binlog from gc.

Reply:
IsLogExists only do some string comparison, should be fast by using RLock.